linchpin/cli/__init__.py

```diff
@@ -119,6 +119,54 @@ def lp_init(self, providers=['libvirt']):
             self.ctx.log_state('Error: {0}'.format(e))
             sys.exit(1)
 
+    def _write_to_inventory(self, tx_id=None, inv_path=None, inv_format="cfg"):
+        latest_run_data = self._get_run_data_by_txid()
+        if tx_id:
+            latest_run_data = self._get_run_data_by_txid(tx_id)
+        all_inventories = []
+        try:
+            for t_id in latest_run_data:
+                targets = latest_run_data[t_id]["targets"][0]
+                # if there are multiple targets mentioned in pinfile
+                # the multiple inventory files are being generated
+                inv_file_count = 0 if len(targets) > 1 else False
+                for name in targets:
+                    try:
+                        lt_data = targets[name]["inputs"]["layout_data"]
```

Comment:
This function should verify that layout_data exists before running this function. It seems this is the cause of that failed test previously. Maybe it's already been fixed, however...